llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -2192,14 +2192,42 @@ findPrologueEndLoc(const MachineFunction *MF) {
   // better off synthesising an early prologue_end.
   auto CurBlock = MF->begin();
   auto CurInst = CurBlock->begin();
-  while (true) {
+
+  // Find the initial instruction, we're guaranteed one by the caller.
+  while (CurBlock->empty())
+    CurInst = (++CurBlock)->begin();
+
```

Comment:
`// Find the initial instruction, we're guaranteed one by the caller.` Can we assert `CurInst != CurBlock->end()` here to check that?